common/src/main/java/dev/schmarrn/lighty/mode/CarpetMode.java

```diff
@@ -60,8 +60,7 @@ public void compute(ClientLevel world, BlockPos pos, BufferBuilder builder) {
 
         int color = LightyColors.getARGB(blockLightLevel, skyLightLevel);
 
-        double offset = LightyHelper.getOffset(up, posUp, world);
-
+        double offset = LightyHelper.getOffset(blockState, blockStateUp);
```

Comment:
just passing blockStateUp in this case when applying the suggestion in LightyHelper would work